pettingzoo/classic/go/go_env.py

```diff
@@ -98,7 +101,11 @@ def _encode_rewards(self, result):
     def observe(self, agent):
         current_agent_plane, opponent_agent_plane = self._encode_board_planes(agent)
         player_plane = self._encode_player_plane(agent)
+
+        self.board_history = np.dstack((current_agent_plane, opponent_agent_plane, self.board_history[:, :, :-2]))
```

Comment:
This logic should go in the step() function rather than the observe function. The reason is that observe can be called many times, or no times, after a step(). In other words, observe() should never change the env state, it should only be a getter. step() should change the env state.